quelpa.el

```diff
@@ -548,9 +555,9 @@ position."
 ;;; Run Process
 
 (defun quelpa-build--run-process (dir command &rest args)
-  "In DIR (or `default-directory' if unset) run COMMAND with ARGS.
+  "In DIR (or symbol `temporary-file-directory' if unset) run COMMAND with ARGS.
 Output is written to the current buffer."
-  (let ((default-directory (file-name-as-directory (or dir default-directory)))
+  (let ((default-directory (file-name-as-directory (or dir temporary-file-directory)))
```

Comment:
You mentioned in the other PR:

> No, the Quelpa cache is stored in quelpa-persistent-cache-file and unaffected by this. The reason I change this is to make quelpa-build--run-process successfully even from remote directory. Git kind of fail to run if pwd is set to some remote directory

By remote directories, do you mean using TRAMP?  I'm confused about how Quelpa would be used in such a way.  Could you explain a little bit about this use case please?

Reply:
Yes, TRAMP or anything that `file-remote-p` return t.
The use case is like:
- Open a remote file
- From there invoke `M-x quelpa-upgrade` or any quelpa command
- The default folder will be set to remote default folder and can make process (git) fail to run.

This scenario can be easily happens especially when you open remote file and realize you want to upgrade or install some packages through Quelpa.

Reply:
I see, thanks.

I may be wrong, but I feel like this isn't the proper way to handle this.  It seems like Quelpa should always run its builds from either a specific, configured directory or a temporary directory (which should probably be cleaned up afterward).  It doesn't seem right for it to ever use the value of `default-directory` in whatever happens to be the current buffer when the user runs the command.

Am I missing something?  What do you think?

Reply:
I agree, note that `default-directory` is significant only when we are invoking external process. In that case the`default-directory` is external process pwd. We are not really read/write anything there, but some external processes are sensitive about where it's running from, git in Windows is an example.

For my change, we will using preconfigured directory or temporary directory to set the pwd.

Reply:
Why wouldn't the git repo directory always be used as the process working directory?

Reply:
I would think that before you have the directory being clone to, you have to run `git clone` from somewhere. At that time, the git repo is not ready yet and we have to run from temporary dir.
After that, we did run from git repo directory now.

Reply:
* Cloning a repo (i.e. calling `git clone`) should always be done from `quelpa-dir`, right?
* Building a package should always be done from `quelpa-build-dir`, right?

If so, it would seem like a bug to ever use any directory other than those, including `default-directory` or a temporary one.

Am I missing something?

Reply:
Are you suggesting to set `default-directory` to `quelpa-dir` (or `quelpa-build-dir`) instead of `temporary-file-directory`? Whichever is fine to me too, as long as the pwd of launched process is not current buffer directory.

Note that `quelpa-dir` just point to general quelpa directory (i.e. `~/.emacs.d/quelpa`), while `quela-build-dir` (pointed to `~/.emacs.d/quelpa/build/`) is actually where the package is cloned into.

Reply:
The working directory should *always* be the correct *Quelpa* directory, not whatever happens to be a buffer's default directory.

Please don't mark conversations as resolved just because I've been busy and haven't responded to your latest message yet.